### PR TITLE
[FIX] * : multi company fixes

### PR DESCRIPTION
**Pull request by Feyensv**
--
I confirm I have signed the CLA and read the PR guidelines at www.odoo.com/submit-pr

### CODE REVIEW

File: odoo/addons/test_new_api/tests/test_company_checks.py [hunk 1]
```diff
@@ -72,3 +72,51 @@ def test_company_check_4(self):
...
+            companies
+        )
+
+        self.assertEqual(
+            user.sudo().with_context(
+                allowed_company_ids=[self.company_b.id, 'abc']).env.company,
+            self.company_b
+        )
+        """
+        wrong_env = user.sudo().with_context(
```
Comment (odony) [inline odoo/addons/test_new_api/tests/test_company_checks.py]: remove?

File: odoo/api.py [hunk 2]
```diff
@@ -541,35 +542,51 @@ def user(self):
...
+        :raise AccessError: invalid or unauthorized 'allowed_company_ids' context key content.
+        :return: current company (default=`self.user.company_id`)
+        :rtype: res.company
+        """
+        company_ids = self.context.get('allowed_company_ids', [])
         if company_ids:
-            company_id = int(company_ids[0])
-            if company_id in self.user.company_ids.ids:
-                return self['res.company'].browse(company_id)
+            if not self.su:
```
Comment (odony) [inline odoo/api.py]: Is this worth a comment to remind us of the rationale for special-casing `su` mode?
The commit message for that change deserves a bit more than a one-liner too :p

**Comment by odony** [comment]
@robodoo r- to give @Feyensv a chance to answer my (minor) comments ;)

File: odoo/addons/test_new_api/tests/test_company_checks.py (hunk 1 above)
Comment (Feyensv) [reply odoo/addons/test_new_api/tests/test_company_checks.py]: It is a highlight to clearly say that it is intended to be so, even if not testable.
If you think it doesn't help, I can remove it.

File: odoo/api.py (hunk 2 above)
Comment (Feyensv) [reply odoo/api.py]: Information added in method docstring.

**Comment by Feyensv** [comment]
@odony if the comments/translations/commit message are good for you, please re-r+ :pray:.

**Comment by Feyensv** [comment]
@robodoo r+

**Comment by odony** [comment]
ok ;)

**Pull request closed without merging**